src/main/java/me/zikani/labs/pakadali/web/qr/QRCodeRoute.java

```diff
@@ -0,0 +1,47 @@
+package me.zikani.labs.pakadali.web.qr;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import me.zikani.labs.pakadali.api.qr.service.impl.QRCodeServiceImpl;
+import me.zikani.labs.pakadali.api.qr.service.interfaces.QRCodeService;
+import me.zikani.labs.pakadali.api.qr.validator.impl.QRCodeValidatorImpl;
+import me.zikani.labs.pakadali.api.qr.validator.interfaces.QRCodeValidator;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+import spark.Spark;
+
+import static java.util.Objects.requireNonNull;
+import static me.zikani.labs.pakadali.PakadaliApplication.MEDIA_TYPE_IMAGE_PNG;
+
+public class QRCodeRoute implements Route {
+    private final Cache<String, byte[]> cache;
+
+    public QRCodeRoute(Cache<String, byte[]> qrCodeCache) {
+        this.cache = requireNonNull(qrCodeCache, "qrCodeCache");
+    }
+
+    @Override
+    public Object handle(Request request, Response response) {
+        final String content = request.queryMap("content").value();
+        QRCodeValidator qrCodeValidator = new QRCodeValidatorImpl();
```

Comment:
Do we really need to create this validator for every request?

Reply:
We can discuss on it.... It can be made static but I don't know how unit tests will be written on the static methods to test the logic in the methods